sycl/test/regression/spirv-broadcast.cpp

```diff
@@ -0,0 +1,105 @@
+// RUN: %clang -std=c++11 -fsycl %s -o %t.out -lstdc++ -lOpenCL -lsycl
+// RUN: env SYCL_DEVICE_TYPE=HOST %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+//==-- spirv-broadcast.cpp -------------------------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// This test checks that __spirv_GroupBroadcast works correctly for one-,
+// two- and three-dimensional work-groups.
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl.hpp>
+#include <algorithm>
+#include <cassert>
+#include <numeric>
+using namespace sycl;
+
+int main() {
+
+  queue q;
+  constexpr int N = 16;
+  std::array<int, N> input;
+  std::array<int, 3> output;
+  std::iota(input.begin(), input.end(), 1);
+  std::fill(output.begin(), output.end(), 0);
+
+  nd_range<1> range1d(N, N);
+  nd_range<2> range2d(range<2>(4, 4), range<2>(4, 4));
+  nd_range<3> range3d(range<3>(4, 1, 4), range<3>(4, 1, 4));
+
+  // 1D test
+  {
+    buffer<int, 1> input_buf(input.data(), input.size());
+    buffer<int, 1> output_buf(output.data(), output.size());
+    q.submit([&](handler &cgh) {
+      auto in = input_buf.get_access<access::mode::read>(cgh);
+      auto out = output_buf.get_access<access::mode::write>(cgh);
+      cgh.parallel_for<class kernel1d>(range1d, [=](nd_item<1> it) {
+        group<1> g = it.get_group();
+        int i = it.get_global_id()[0];
+#ifdef __SYCL_DEVICE_ONLY__
```

Comment:
Not yet, but there will be one very soon.  I'm working on an implementation of the extension described in https://github.com/intel/llvm/pull/1079, which includes a `sycl::broadcast` function wrapping this  SPIR-V intrinsic.

I submitted a separate PR for this change because it's a relatively simple fix and doesn't depend on the extension specification being approved.

Reply:
I didn't realize that when I asked for the test.
It would be better to validate `sycl::broadcast` function.
Are you going to update/replace/drop/leave-as-it it later? I'm okay with any option :-).

Reply:
Ah, ok.  I'll drop this test for now, then.  I'll be adding a separate test for `sycl::broadcast` that looks a lot like this.